adrs/3122196229/README.md

```diff
@@ -0,0 +1,55 @@
+# NodeJS files and directories names conversion
+
+* **State:** Draft
+* **Created:** 2020-11-10
+* **Tags:** nodejs
+
+## Context
+
+Across the ecosystem, we can find all sort of files and directories naming
+conversions in NodeJS projects.
+
+That has caused misalignment, shifting the focus away from more important
+matters, also have created conventions that introduce problems such:
+
+* Disagreement around English language, making it even more difficult for the
+  non-English speaker to figure out the right decision.
+* Case sensitive vs. case insensitive file systems, causing issues in different
+  environments such as your local computer vs. CI/CD computers.
+* Format mismatching around separating English words.
+
+The `camelCase`, and/or `PascalCase` for naming convention, although this seems
+a good solution there are some caveats.
+
+What happen for the following cases `httpClient` vs. `HTTPClient` vs.
+`HttpClient`?
+
+* Which one is the more appropriate choice here without getting into
+  disagreement and use the phrase "it depends"?
+
+In some cases, the file `PascalCase` is used if the file is a class or
+a React component, otherwise follow `camelCase` is used.
+
+* What happen when the file contains more than a class or a React component? How
+  do we take the appropriate decision about it without getting into
+  disagreement and use the phrase "it depends"?
+
+Also, does that means that if we refactor the code, and we move the React
+component out of a file, that file name must change?
+
+That would introduce more potential refactoring like renaming file paths,
+adding more unnecessary line of code changes.
+
+As well as issues in `CI` environments, where it may be the case that `git`
+didn't rename the file due to case-insensitive file systems, and you must use
+`git mv [old file] [new file]` to fix such issue.
+
+## Resolution
+
+File or directory names:
+
+* Should not contain any leading or trailing spaces
+* Must be lowercase i.e., no uppercase or mixed case names are allowed
+* Should separate what constitute a English word by `-`
```

Comment:
This seems arbitrary. Additionally adds cognitive load when trying to translate a function to the actual file. 

IE: `apiRequest` -> `api-request.ts`
vs `apiRequest` -> `apiRequest.ts`

Reply:
I am not sure why `api request` would be the words to describe the content of your file, but if that is the case 🤷🏻   I would reconsider doing such practices since a function renaming means a file change.

That being said, we are entering another realm here about the usage of files and folders rather than naming.

Reply:
I agree, naming your files the same as the function is generally bad practice. You can find it in the wild and I am also guilty of it but files are a form of encapsulation so their name should represent the higher level concept that encapsulates the functionality. This is why it _is_ okay to name files after class names, because classes are also a form of encapsulation so their names already reflect the higher level concept.

That being said, when it comes to naming files, just like with naming classes, cognitive load is required to discover the appropriate name that represents the encapsulation.

Reply:
In the vast majority of cases, components are named the same as their file and it would make the codebase harder to traverse otherwise.

Reply:
Sure,

Let's bring the conversation around naming since we are entering something completely different from what the ADR is trying to solve right now.

Reply:
Fair enough, I wasn't considering functional components. My feeling there is that while they technically are functions, they are spiritually an encapsulation of the state, logic, and sub-components that they render. This is the same encapsulation that classes are typically used for. That is reflected by the fact that typically functional components are named the same way as class components. I think it follows that a file housing a functional component can be named after the component.